llvm/include/llvm/Support/PropertySetIO.h

```diff
@@ -91,26 +138,29 @@ class PropertyValue {
 
   Type getType() const { return Ty; }
 
-  size_t size() const {
+  SizeTy size() const {
     switch (Ty) {
     case UINT32:
       return sizeof(Val.UInt32Val);
+    case BYTE_ARRAY:
+      return getRawByteArraySize();
     default:
       llvm_unreachable_internal("unsupported property type");
     }
   }
 
 private:
   template <typename T> T &getValueRef();
+  void copy(const PropertyValue &P);
 
   Type Ty = NONE;
   union {
     uint32_t UInt32Val;
+    // Holds first sizeof(size_t) bytes of size followed by actual raw data.
+    byte *ByteArrayVal;
```

Comment:
I'd prefer byte* here, as it is more efficient - no need in temp buffers when encoding, and simple enough. Do you have particular concerns about memory non-safety?

Reply:
The main concern is maintainability of manual memory allocation/deallocation. Can we use std::unique_ptr?

Reply:
std::unique_ptr inside a union will actually lead to more complex and less reliable code. If it was c++17, I'd use variant with unique_ptr, and it would be maintainable. So how about I add a TODO that with C++17 use variant and unique_ptr?

Reply:
I'm okay with that. 

It might be worth considering another refactoring of PropertyValue and add support for arrays in derived class to separate methods specific to arrays.

Reply:
Yes, maybe. But this is out of scope for this PR.